### ISSUE TITLE
Email Notifications Fail When Domains Lack Scheme

### ISSUE THREAD

**Issue by Nicholas-Westby** [nathanwoulfe/Plumber#17]
Umbraco 7.7.9. Plumber 0.5.2.

I wasn't getting notification emails on workflow events (e.g., request for approval to publish), and I noticed this in my error log:

```
 2018-02-05 17:40:51,249 [P11392/D7/T19] ERROR Workflow.Helpers.Notifications - Error sending notifications
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at Workflow.Helpers.UrlHelpers.GetFullyQualifiedEditUrl(String partialUrl)
   at Workflow.Helpers.Notifications.Send(WorkflowInstancePoco instance, EmailType emailType)
```

Seems like the site URL configuration was the source of the issue:

![Site URL's](https://user-images.githubusercontent.com/5933222/35836443-781f1256-0a9d-11e8-8e06-1d295cc53721.png)

I had configured, for example, "mysite.localhost", which seems to match what is indicated in the description (i.e., "www.domain.com"). Once I changed that to "http://mysite.localhost/", email notifications started working.

I'm thinking one or both of these could happen:

* The descriptive text could mention the scheme (e.g., "http://").
* The code could parse the URL regardless of whether or not there is a scheme (I imagine "http://" would be assumed in that case).

**Comment by nathanwoulfe** [comment]
Nice catch, will implement both.

**Comment by nathanwoulfe** [comment]
This was resolved in 0.6 -> will fallback to the scheme matching the current request if none present in the settings value (for the site URL), and fall back to https for edit
